pub_key.go

```diff
@@ -15,7 +18,77 @@ import (
 // An address is a []byte, but hex-encoded even in JSON.
 // []byte leaves us the option to change the address length.
 // Use an alias so Unmarshal methods (with ptr receivers) are available too.
-type Address = cmn.HexBytes
+type Address struct {
+	cmn.HexBytes
+}
+
+func (a *Address) UnmarshalJSON(data []byte) error {
+	readable, deserialized, err := bech32cosmos.Decode(string(data))
+	if err != nil {
+		return err
+	}
+	if strings.ToUpper(readable) != "CSMSADDR" {
```

Comment:
Mh, I'm wondering whether we want to enforce the address in go-crypto. I can imagine that other zones would like to use a very similar bech encoding scheme. We should allow users to define their own `readable`.

Reply:
So where does the readable value get injected from?

Reply:
Think it makes sense to unmarshal the readable and the data into the `Address` struct. Then consumers of go-crypto can enforce what they need to about the addresses